src/pytorch_lightning/tuner/lr_finder.py

```diff
@@ -380,10 +392,20 @@ def on_train_batch_end(
         # Check if we diverging
         if self.early_stop_threshold is not None:
             if current_step > 1 and smoothed_loss > self.early_stop_threshold * self.best_loss:
-                trainer.fit_loop.max_steps = current_step  # stop signal
+                trainer.should_stop = True  # stop signal
                 if self.progress_bar:
                     self.progress_bar.close()
 
+        if trainer.should_stop:
+            self.should_stop_rank = int(
+                trainer.strategy.reduce(
+                    torch.tensor(trainer.global_rank, device=pl_module.device), reduce_op="max"
+                ).item()
```

Comment:
I don't understand how this can work. The result of this will always be the "highest" rank: `max(0, 1, 2, ...., n) = n`

Let's simplify this (see also my comment above about broadcast):

I suggest to reduce using sum. Then, if any item is 1 (should_stop), the sum will be > 0. And you won't need the extra broadcast below because every rank already will have the same number. You can just set 

trainer.should_stop = ...reduce(...).item() > 0

Then, when stopped, you broadcast the "results" from rank 0 to all others. This should be good.

Reply:
I am trying to broadcast the rank which is triggering the should_stop. If rank 1 triggers should_stop, rank 1 results should be broadcasted. If rank 1 and 2 triggers should_stop together (there's a barrier), we populate results from any of those ranks.